garak/_plugins.py

```diff
@@ -123,7 +124,12 @@ def load_plugin(path, break_on_fail=True) -> object:
             return False
 
     try:
-        plugin_instance = getattr(mod, plugin_class_name)()
+        from garak.configurable import Configurable
+
+        if issubclass(getattr(mod, plugin_class_name), Configurable):
+            plugin_instance = getattr(mod, plugin_class_name)(config_root=config_root)
+        else:
+            plugin_instance = getattr(mod, plugin_class_name)()
     except AttributeError as ae:
         logging.warning(
             "Exception failed instantiation of %s.%s", module_path, plugin_class_name
```

Comment:
This can be simpler as all supported `plugin` types are `Configurable` (this was not the case when I made this edit), we should be able to fall back to just:
``` Suggestion
    try:
        plugin_instance = getattr(mod, plugin_class_name)(config_root=config_root)
    except AttributeError as ae:
        logging.warning(
            "Exception failed instantiation of %s.%s", module_path, plugin_class_name
```

Reply:
Might be worth adjusting this to check that `config_root` is a valid parameter for the class though, maybe:
``` Suggestion
    try:
        if "config_root" not in inspect.signature(self.__init__).parameters:
            raise AttributeError('Incompatible function signature: "config_root" is incompatible with this plugin')
        plugin_instance = getattr(mod, plugin_class_name)(config_root=config_root)
    except AttributeError as ae:
        logging.warning(
            "Exception failed instantiation of %s.%s", module_path, plugin_class_name
```

Reply:
This is good, yes. I now think the warning message logged is a vague compared to what's actually caught, but this revision looks like good logic.